cli/tests/services.bats

```diff
@@ -153,6 +160,114 @@ EOF
 
 # ---------------------------------------------------------------------------- #
 
+# bats test_tags=services:restart
+@test "restart: errors before restarting if any service doesn't exist" {
+  export FLOX_FEATURES_SERVICES=true
+  setup_start_counter_services
+
+  run "$FLOX_BIN" activate --start-services -- bash <(cat <<'EOF'
+    source "${TESTS_DIR}/services/register_cleanup.sh"
+    "$FLOX_BIN" services restart one two invalid
+EOF
+)
+  assert_failure
+  assert_output --partial "❌ ERROR: Service 'invalid' not found"
+
+  run cat start_counter.one
+  assert_output "1"
+  run cat start_counter.two
+  assert_output "1"
+  run cat start_counter.sleeping
+  assert_output "1"
+}
+
+# bats test_tags=services:restart
+@test "restart: restarts a single service" {
+  export FLOX_FEATURES_SERVICES=true
+  setup_start_counter_services
+
+  run "$FLOX_BIN" activate --start-services -- bash <(cat <<'EOF'
+    source "${TESTS_DIR}/services/register_cleanup.sh"
+    "$FLOX_BIN" services restart one
+EOF
+)
+  assert_success
+  assert_output --partial "✅ Service 'one' restarted"
+
+  run cat start_counter.one
```

Comment:
nonblocking(unless we think we'll hit the race): restart isn't blocking, is it? Then I think this has a race: `restart` and `cat` finish before the service actually finishes restarting. If so I think this needs some explanation of why we don't think we'll hit the race or to handle it

Reply:
I got 100 successful runs out of it with:
```sh
for i in {1..100}; do echo $i; just integ-tests services.bats -- -f restart || break; done
```

However, now that you mention it, I suspect our `register_cleanup.sh` is giving it enough time to fall in sync. Which will cause us problems when we eventually remove that in favour of just relying on the watchdog.

I'm unsure how we can make it more reliable though. I don't really want to add a bare `sleep` or load bearing `flox services status`. It's easier for the positive cases, where we expect a restart, because we can poll with a timeout until we get the value that we expect. But for the negative cases, where we don't expect a restart, we'd have to poll for a fixed amount of time to ensure the value hasn't changed. Thoughts?

(nb: `process-compose process list` output help us here because `restarts` doesn't show manually invoked restarts and `age` doesn't have anything for Completed services)

Reply:
Some ideas that @ysndr and I explored this morning:

1. Use pipes to co-ordinate the tests with the services starting like we've just done for the logging tests. This only helps the positive cases and doesn't help the negative cases. 
1. Polling for the `process-compose` socket being deleted as a guard against the implicit dependency on `register_cleanup.sh` being removed. This still doesn't give us any actual guarantee that the services have or haven't finished starting before `process-compose` was shutdown though.
1. Don't make assertions on the negative cases where we don't restart anything for a misnamed service. I wouldn't feel good about removing this, even though it's unit tested by `processes_by_name_or_default_to_all`, because it could be easy to regress.
1. Don't make assertions on the negative cases where we restart named services. I feel kinda-OK about doing this because either us or `process-compose` would have to do something pretty bad in the argument parsing to restart more services than we asked for.

I'm going to think about it some more while I implement the rest of the functionality. Other ideas welcome!

Reply:
Yeah we could poll for the positive cases. For the negative cases, we wouldn't ever have a test flake, we just wouldn't necessarily detect a regression. If we comment and check for the lack of restart after we've polled for the positive cases restarting, I think that's low enough odds that I'm okay with that gamble. I don't think we're too likely to regress. We could also add a --refute-output "Service 'whatever' restarted" if we're worried about regressing what we're passing to process-compose

Reply:
Yeah I don't feel great about testing negative cases like this. There's not an easy way to differentiate between "it didn't happen before a timeout" and "it's never going to happen," and you'll end up just needing to wait for the entire timeout when things are working properly. I think it's enough to unit-test the "is there an error" functionality.

Reply:
I've added a poll and timeout for the positive cases.

I've left the negative cases as-is for now, with a comment on the test that doesn't restart anything, because I think it's still valuable to have them if they don't produce false failures.